TreeCrownDetection/MainWindow.xaml.cs

```diff
@@ -269,96 +259,90 @@ private async void SaveFileButton_OnClick(object sender, RoutedEventArgs e)
             {
                 DefaultExt = ".geoTIFF",
                 Filter =
-                    "TIF Files (*.tif)|*.tif|geoTIFF Files(*.geoTIFF)|*.gTIFF|TIFF Files (*.tiff)|*.tiff"
+                    "ESRI Shapefile (*.shp)|*.shp|geoTIFF Files(*.geoTIFF)|*.gTIFF"
             };
 
-            bool? result = dlg.ShowDialog();
+            var result = dlg.ShowDialog();
             if (result != true) return;
 
 
-            Console.WriteLine(String.Format("Saving to file: {0}", dlg.FileName));
-            await SaveToFile(dlg.FileName, true);
-
+            Console.WriteLine("Saving to file: {0}", dlg.FileName);
+            await SaveToFile(dlg.FileName);
         }
 
-        private async Task SaveToFile(String FileName, bool shapefile)
+        private async Task SaveToFile(string fileName)
         {
-            String FileNameGTiff = null;
+            var fileExtension = fileName.Split('.').ToArray().Last();
+            var shapefile = fileExtension == "shp";
```

Comment:
imho, better approach would be to pass by argument if it is shapefile than resolving it inside saving function.

Reply:
You have to parse user input somewhere, it does not mater in with layer. In this approach you are not limited to two different save file formats.